### PR TITLE
Refresh Token functionality & IIRC Bot

### PR DESCRIPTION
**Pull request by bryancua**

### CODE REVIEW

File: bot-command.go [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+// IIRCBotMessage is a type of IIRCBotCommand which sends a specific message through the
+// bot connection.
+type IIRCBotMessage struct {
+	bot  *IIRCBot
+	Cmd  string
+	Args string
+}
+
+func (m *IIRCBotMessage) String() string {
+	return fmt.Sprintf("[Mesasge] %s %s", m.Cmd, m.Args)
```
Comment (tmathews) [inline bot-command.go]: Typo?

File: bot.go [hunk 2]
```diff
@@ -0,0 +1,306 @@
...
+			if b.needsSendPong() {
+				if err := b.send(CmdPong, ":tmi.twitch.tv"); err != nil {
+					b.Log(logger.SeverityError, "Could not send pong. "+err.Error())
+				}
+			}
+
+			// Run a queue command if there are any. 
+			if len(b.Queue) > 0 {
+				cmd := <-b.Queue
+				if err := cmd.Run(); err != nil {
```
Comment (tmathews) [inline bot.go]: I find it kinda awkward that commands store a reference to the bot when the bot is always in control of commands. It's an unneeded reference. Isn't it better to not convolute the command structures with the extra reference when the executing bot should be in control anyways and pass itself to `Run`? Binding bot to the command makes it less flexible for command reuse no and future adaptability, no?

Like `cmd.RunWithBot(b)`.

**Review by tmathews — APPROVED** [review]

File: bot-command.go (hunk 1 above)
Comment (bryancua) [reply bot-command.go]: yes thanks

File: bot.go (hunk 2 above)
Comment (bryancua) [reply bot.go]: YES. OF COURSE. I will do this.

**Comment by tmathews** [comment]
Good to go, thanks.

**Pull request merged**